front/app/containers/IdeasNewSurveyPage/index.tsx

```diff
@@ -71,9 +71,16 @@ const IdeasNewSurveyPage = () => {
       authUser: authUser?.data,
     });
 
+  // Hard to understand why this is needed without context.
+  // The phase id checks are also unclear.
+  // Please replace this text and add a comment if you know.
   const userCannotViewSurvey =
     !canModerateProject(project.data.id, authUser) &&
-    phase_id !== currentPhase?.id;
+    /* Something I could deduct: when this code was added, we made the (wrong) 
+    assumption that `phase_id` was always a string (we were type casting the phase_id param). 
+    So I _think_ we are checking here whether the phase_id from the search params is the same
+    as currentPhase?.id, when both are strings.*/
+    phaseIdFromSearchParams !== currentPhase?.id;
```

Comment:
I realize that with `phaseIdFromSearchParams` we are now potentially checking `null !== currentPhase?.id` instead of `undefined !== currentPhase?.id`. However, I think it's fine. See comment above the line.

Reply:
!== already checks that. I will change it back to undefined in case it's null for now so I don't have to worry about it in this PR.